### DISCUSSION TITLE
Search for a particular product

### DISCUSSION THREAD

**Discussion by remi-braun** [CS-SI/eodag#327] [Q&A]
Hello,

I haven't found it in the documentation, but is there a way to find a particular product accross the providers ?

My usecase is:

I query product (let's name it A) on a provider, but A is offline.
I want to try if A is online on a provider of my list of registered providers, so I want to query A specifically.

For now I try to narrow the search down with the product_type, start+end time and a geometry, but I think this is rather complicated and maybe error-prone.

**Comment by maximlt** [comment]
How about looking for a product's ID?

https://eodag.readthedocs.io/en/latest/notebooks/api_user_guide/4_search.html#id-and-provider

**Reply by remi-braun** [reply]
Those IDs are provider-agnostic or are there some known limitations ?

**Reply by remi-braun** [reply]
Just a quick remark, `self.search(id=prod.properties["id"])` doesn't work although `self.search(id=prod.properties["title"])` does 😓

**Reply by remi-braun** [reply]
I will precise what I wrote hereover: 
This is for eodag-sentinelsat.
The search result for this plugin has as properties:
- for `id` something else than the name: `5f7e3737-1b86-4017-a9f1-42a91a0cfe3d` 
- for `title` what should come from the name

Moreover, the search by id doesn't work

It seems to have been fixed with the new metadata mapping but not released yet (can't wait for this release 😛)
Sorry for this monologue (again !)

**Reply by sbrunato** [reply]
yes, that's it. The `id` is a property that sometimes points to an `uuid` or to a `title`. 
To identify a product across providers, we use the property `title`. And this is the property that has to be passed through the `id` query-argument like you wrote:
```py
self.search(id=prod.properties["title"])
```
This behavior is not very obvious and we'd have to enhance metadata-mapping to go to something like:
- `p.properties["title"] = p.properties["id"] = "S2B_MSIL1C_20210427T103619_N0300_R008_T31TCJ_20210427T124539"`
- `p.properties["uuid"] = "5f7e3737-1b86-4017-a9f1-42a91a0cfe3d"`

**Comment by sbrunato** [comment]
> I query product (let's name it A) on a provider, but A is offline.
> I want to try if A is online on a provider of my list of registered providers, so I want to query A specifically.

For your use-case, I'd use the following tools:
- [dag.available_providers()](https://eodag.readthedocs.io/en/latest/notebooks/api_user_guide/2_providers_products_available.html?highlight=available_providers#Providers-available) to get available providers
- `product.properties["storageStatus"]` to check if a product is online or not

And if a given product is not `ONLINE`, then loop search by id over available providers:
```py
dag.search(id=product.properties["title"], provider="next_available_provider")
```

**Reply by remi-braun** [reply]
Oh the `next_available_provider` is very nice ! I did the work by hand before (determining available provider, sorting them by priority...) 
This will simplify everything 🥳

**Reply by sbrunato** [reply]
> Oh the next_available_provider is very nice !

oops it is not a special key, it could have been `"foo_provider"`... Sorry you'll have to use `dag.available_providers()` :grimacing:

**Reply by remi-braun** [reply]
Haha okay, nevermind 😄
